### ISSUE TITLE
Album Art update fails from last.fm

### ISSUE THREAD

**Issue by dsiminiuk** [airsonic-advanced/airsonic-advanced#478]
## Problem description

When browsing the audio library, identify an artist and album that has no cover art. Click "Change', select an image from last.fm, UI reports "Image downloaded."

Click "Back", the downloaded image is not shown. Refreshing the browser makes no difference.

## System information

 * **Airsonic version**: 11.0.0-SNAPSHOT.20210210174643
 * **Operating system**: Ubuntu 20.04.2 LTS.
 * **Java version**: openjdk version "11.0.10" 2021-01-19
 * **Proxy server**: None
 * **Client**: Vivaldi | 3.6.2165.3
 * **Language**: US-EN

## Additional notes

This issue has existed regardless of various browser updates or airsonic updates, not new.

**Comment by dsiminiuk** [comment]
This still isn't working for me in 11.0.0-SNAPSHOT.20210214115559.

Updated to 11.0.0-SNAPSHOT.20210221091539 just now, same.

To reproduce...

Example, go to Artist: Alanis Morisette Album: Jagged Little Pill - album art is the typical color panel with album name in random colored background with the artist's name on a black background block below. Click on the album image and you see a larger version of the same image.

Click "Change" - choose an album cover in the array of selections by clicking on it.

"Image downloaded." is displayed.

Click "< Back"

You see the same original default image as before. A browser refresh does not display the downloaded image.

If you click on the image you get the larger version of the same default image, not the image that supposedly downloaded.

As though it did nothing at all.

Ubuntu 20.04.1 LTS
openjdk version "11.0.10" 2021-01-19
11.0.0-SNAPSHOT.20210221091539

**Comment by dsiminiuk** [comment]
@randomnicode can you please reopen this issue?

**Comment by randomnicode** [comment]
Unable to reproduce this locally.

Start with a random sample album:
![image](https://user-images.githubusercontent.com/45723468/108905111-2d645a00-75d4-11eb-9719-dd784d3157ee.png)

Click on Change and this screen comes up:
![image](https://user-images.githubusercontent.com/45723468/108905257-57b61780-75d4-11eb-8a34-5188111e2873.png)

Selecting the first Image and this familiar Image Downloaded comes up:
![image](https://user-images.githubusercontent.com/45723468/108905381-7fa57b00-75d4-11eb-94a7-99c297e899f9.png)

And then when you press back, the image appears.
![image](https://user-images.githubusercontent.com/45723468/108905511-a499ee00-75d4-11eb-9a57-56238102c4d5.png)


What might be helpful is:
- Checking your cover art source (perhaps it's set to embedded tag only) in Settings -> General
- Starting with an album with no cover in its folder,
  1. Note its last modified date (outside Airsonic)
  2. Select the album cover art as before in Airsonic
  3. Notice if an image file has downloaded in that folder (cover.jpg or cover.png etc.)
  4. Note the folder's last modified date (outside Airsonic) -> this should have changed from (i) to something recent

**Comment by dsiminiuk** [comment]
@randomnicode thanks for the assist.

Settings-> General-> Cover art source: Prefer embedded tag over external

I set it to external only.

I went back to Alanis Morisette as in my original test and now I see the cover art in Airsonic and there is a cover.jpg in the folder on the file system dated 2/21. YAY!

I must have an invalid image embedded in the file that is overwriting the local file.

I can live with this solution.

Thank you!!

**Comment by dsiminiuk** [comment]
@randomnicode 

Hmmm. Not exactly right yet.

If I choose another album and download the cover art, and go back 9to the album main page0, the image has not updated. The cover.jpg file was downloaded and present in the folder. However, now if I click on the fake art I see the downloaded image.

Browser refresh and logging out/in doesn't change it.

I hope that makes sense.

**Comment by jobenvil** [comment]
Kind of caching somehow. I have similar issue. I see the updated time on the unix folder and I see the new cover:
```
desigual@arkanoid:/media/sda/album/musica/kinder/Andreas Steinhöfel - Rico, Oskar und die Tieferschatten/Rico, Oskar und die Tieferschatten - CD01$ ll
total 113736
-rw-rw-r-- 1 desigual sambashare 17766400 may 18  2016 '01-Samstag - Die Fundnudel 01.mp3'
-rw-rw-r-- 1 desigual sambashare 19853312 may 18  2016 '02-Samstag - Die Fundnudel 02.mp3'
-rw-rw-r-- 1 desigual sambashare 19019776 may 18  2016 '03-Immer noch Samstag - Oskar 01.mp3'
-rw-rw-r-- 1 desigual sambashare 18503680 may 18  2016 '04-Immer noch Samstag - Oskar 02.mp3'
-rw-rw-r-- 1 desigual sambashare 10268672 may 18  2016 '05-Das Ferientagebuch 01.mp3'
-rw-rw-r-- 1 desigual sambashare 12894208 may 18  2016 '06-Montag - Der Bühl 01.mp3'
-rw-rw-r-- 1 desigual sambashare 17979392 may 18  2016 '07-Montag - Der Bühl 02.mp3'
-rw-rw-r-- 1 desigual sambashare   168261 feb 24 16:04  cover.png

```
When I press back, the image doesn't appear BUT if later on I click on "zoom", the downloaded picture appears. When I return in chrome, no image appears. When I press "Home" the downloaded image is there. If I click again in the downloaded picture, I get again no picture in the album view. Since I was using reverse proxy I thought it could be a caching issue, but I connected directly an I have the same behaviour. I will try to make a short video.

**Comment by jobenvil** [comment]
Due to GitHub limit exceeded I have to put a download [link](https://cloud.hiperborea.de/f/1e43a7069b0e450391e0/?dl=1) instead an animated gif. I will delete the video after @randomnicode have had a look on it.

**Comment by dsiminiuk** [comment]
I log into airsonic from two different PCs on the same network and I see the cover art on my laptop but not on my desktop. Logged in using the admin user, same browser - Vivaldi 3.6.2165.40.

Maybe this is a clue.

**Comment by dsiminiuk** [comment]
Video here https://cloud.siminiuk.com/s/93cQYLz5o9RTCcS

**Comment by jobenvil** [comment]
> And on my desktop I can see the album art in the zoom view but not in the album view.
Yes, I had exactly the same behaviour, but on the exposed video when I clicked on "zoom" the album art is not displayed. This confused me and tried again and again with the zoom view and I preshed CTR+F5 but the album art still not displayed. Strange.

When I try with nginx proxy the home view doesn't show the art cover but the album view and zoom...

My settings had prefered external file over embebed tag for cover art source since beginning.

**Comment by dorel14** [comment]
Hello 
Don't know if it is the same proble bt when i try to download a cover i have this error
`Erreur de téléchargement de l'image.
javax.net.ssl.SSLException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty`


Version | 11.0.0-SNAPSHOT.20210221091539
Serveur | Apache Tomcat/9.0.41, java 11.0.6, Linux (127,2 MB / 200,7 MB)
NGINX as reverse Proxy

**Comment by dorel14** [comment]
Hi 
On debian 9 stretch these 2 commands work for me 
`sudo dpkg --purge --force-depends ca-certificates-java

sudo apt-get install ca-certificates-java'

**Comment by dsiminiuk** [comment]
Did not fix for me. :(

After selecting a new cover art image, the zoomed image was updated, the thumbnail was updated in the artist view, but not the album view.
